src/NuGet.Clients/PackageManagement.UI/Services/INuGetLockService.cs

```diff
@@ -16,6 +17,10 @@ public interface INuGetLockService
         /// </summary>
         bool IsLockHeld { get; }
 
+        JoinableTask<T> EnterNuGetOperation<T>(Func<System.Threading.Tasks.Task<T>> execute, CancellationToken token);
+
+        JoinableTask EnterNuGetOperation(Func<System.Threading.Tasks.Task> execute, CancellationToken token);
+
```

Comment:
This is a public interface. Is it in an assembly that other folks can reference, such that this is a significant breaking API change?